node/heartbeat/monitor.go

```diff
@@ -381,9 +381,30 @@ func (m *Monitor) computeAllHeartbeatMessages() {
 	m.appStatusHandler.SetUInt64Value(core.MetricConnectedNodes, uint64(counterConnectedNodes))
 }
 
+func (m *Monitor) computeInactiveHeartbeatMessages() {
+	m.mutHeartbeatMessages.Lock()
+	inactiveHbChangedMap := make(map[string]*heartbeatMessageInfo)
+	for key, v := range m.heartbeatMessages {
+		isActive := v.GetIsActive()
+		if isActive {
+			continue
+		}
+
+		peerType, shardId := m.computePeerTypeAndShardID([]byte(key))
+		if v.peerType != peerType || v.computedShardID != shardId {
+			v.UpdateShardAndPeerType(shardId, peerType)
+			inactiveHbChangedMap[key] = v
+		}
+	}
+
+	m.mutHeartbeatMessages.Unlock()
+	go m.SaveMultipleHeartbeatMessageInfos(inactiveHbChangedMap)
```

Comment:
this is called from a function that is called GetHeartbeats. It does saving. Same as what happens inside computeAllHeartbeatMessages function

Reply:
fixed